### PR TITLE
Chore: Copy `cortex/util/math` into Loki

### PR DESCRIPTION
**Pull request by kavirajk**
**What this PR does / why we need it**:
Copy the `cortex/util/math` dependency into Loki.
Also moved some helpers from `util/math.go` -> `util/math` package.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: pkg/util/math/rate.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	alpha    float64
+	interval time.Duration
+
+	mutex    sync.RWMutex
+	lastRate float64
+	init     bool
+}
+
+func NewEWMARate(alpha float64, interval time.Duration) *EwmaRate {
```
Comment (cyriltovena) [inline pkg/util/math/rate.go:22]: Is this used somewhere ? Not sure it is.

Comment (kavirajk) [reply pkg/util/math/rate.go]: Yea you are right. Removed it.

**Review by cyriltovena — APPROVED** [review]
LGTM

**Review by sandeepsukhani — CHANGES_REQUESTED** [review]
There are some unwanted spaces under imports sections so can you please fix it?

File: pkg/storage/chunk/gcp/bigtable_object_client.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	otlog "github.com/opentracing/opentracing-go/log"
 	"github.com/pkg/errors"
 
-	"github.com/cortexproject/cortex/pkg/util/math"
+	"github.com/grafana/loki/pkg/util/math"
```
Comment (sandeepsukhani) [inline pkg/storage/chunk/gcp/bigtable_object_client.go:13]: One more here. Sorry for being picky, we would miss fixing it until we start working on it again.

File: pkg/storage/chunk/util/util.go [hunk 3]
```diff
@@ -9,7 +9,7 @@ import (
 
 	ot "github.com/opentracing/opentracing-go"
 
-	"github.com/cortexproject/cortex/pkg/util/math"
+	"github.com/grafana/loki/pkg/util/math"
```
Comment (sandeepsukhani) [inline pkg/storage/chunk/util/util.go:13]: same here

File: pkg/storage/stores/shipper/indexgateway/gateway_test.go [hunk 4]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
 
-	util_math "github.com/cortexproject/cortex/pkg/util/math"
+	util_math "github.com/grafana/loki/pkg/util/math"
```
Comment (sandeepsukhani) [inline pkg/storage/stores/shipper/indexgateway/gateway_test.go:11]: same here

File: pkg/storage/stores/shipper/util/queries.go [hunk 5]
```diff
@@ -5,7 +5,7 @@ import (
 	"sync"
 	"unsafe"
 
-	util_math "github.com/cortexproject/cortex/pkg/util/math"
+	util_math "github.com/grafana/loki/pkg/util/math"
```
Comment (sandeepsukhani) [inline pkg/storage/stores/shipper/util/queries.go:9]: same here

**Review by sandeepsukhani — CHANGES_REQUESTED** [review]
last 4 changes required, I promise :grimacing:

**Comment by kavirajk** [comment]
Fixed the spacing issue @sandeepsukhani. Thanks for the catch :) (thought `goimports -local "github.com/grafana/loki"` should have grouped correctly with removing spaces between those imports. But apparently didn't.

**Review by sandeepsukhani — APPROVED** [review]
LGTM. Thanks Kavi for taking care of changes!

**Pull request merged**